vizro-ai/src/vizro_ai/_vizro_ai.py

```diff
@@ -76,18 +78,29 @@ def _run_plot_tasks(
 
         pass_validation = validated_code_dict.get("debug_status")
         code_string = validated_code_dict.get("code_string")
-        business_insights, code_explanation = None, None
 
-        if explain and pass_validation:
-            business_insights, code_explanation = self._lazy_get_component(GetCodeExplanation).run(
-                chain_input=user_input, code_snippet=code_string
+        if not pass_validation:
+            raise DebugFailure(
+                "Chart creation failed. Retry debugging has reached maximum limit. Try to rephrase the prompt, "
+                "or try to select a different model. Fallout response is provided: \n\n" + code_string
             )
-
-        return {
-            "business_insights": business_insights,
-            "code_explanation": code_explanation,
-            "code_string": code_string,
-        }
+        else:
```

Comment:
No need for this `else` here (I'm surprised if the linter doesn't point this out actually) since the `if` clause raises an exception.